packages/turf-center-median/index.js

```diff
@@ -0,0 +1,124 @@
+ * Press, 2009, 150–151.
+ *
+ * Turfjs has four different functions for calculating the center of a set of
+ * data. Each is useful depending on circumstance.
+ *
+ * `@turf/center` finds the simple center of a dataset, by finding the
+ * midpoint between the extents of the data. That is, it divides in half the
+ * farthest east and farthest west point as well as the farthest north and
+ * farthest south.
+ *
+ * `@turf/center-of-mass` imagines that the dataset is a sheet of paper.
+ * The center of mass is where the sheet would balance on a fingertip.
+ *
+ * `@turf/center-mean` takes the averages of all the coordinates and
+ * produces a value that respects that. Unlike `@turf/center`, it is
+ * sensitive to clusters and outliers. It lands in the statistical middle of a
+ * dataset, not the geographical. It can also be weighted, meaning certain
+ * points are more important than others.
+ *
+ * `@turf/center-median` takes the mean center and tries to find, iteratively,
+ * a new point that requires the least amount of travel from all the points in
+ * the dataset. It is not as sensitive to outliers as `@turf/center`, but it is
+ * attracted to clustered data. It, too, can be weighted.
+ *
+ * @name centerMedian
+ * @param {FeatureCollection<Point>} features Feature Collection of GeoJSON Points
+ * @param {Object} [options={}] Optional parameters
+ * @param {string} [options.weight] the property name used to weight the center
+ * @param {number} [options.tolerance=0.001] the difference in distance between candidate medians at which point the algorighim stops iterating.
+ * @param {number} [options.counter=10] how many attempts to find the median, should the tolerance be insufficient.
+ * @returns {Feature<Point>} The median center of the collection
+ * @example
+ * var points = turf.points([[0, 0], [1, 0], [0, 1], [5, 8]]);
+ * var medianCenter = turf.centerMedian(points);
+ *
+ * //addToMap
+ * var addToMap = [points, medianCenter]
+ */
+function centerMedian(features, options) {
+    // Optional params
+    options = options || {};
+    if (!isObject(options)) throw new Error('options is invalid');
+    var counter = options.counter || 10;
+    if (!isNumber(counter)) throw new Error('counter must be a number');
+    var weightTerm = options.weight;
+
+    // Calculate mean center:
+    var meanCenter = centerMean(features, {weight: options.weight});
+
+    // Calculate center of every feature:
+    var centroids = featureCollection([]);
+    featureEach(features, function (feature) {
+        centroids.features.push(centroid(feature, {weight: feature.properties[weightTerm]}));
+    });
+
+    centroids.properties = {
+        tolerance: options.tolerance,
+        medianCandidates: []
+    };
+    return findMedian(meanCenter, [0, 0], centroids, counter);
+}
+
+/**
+ * Recursive function to find new candidate medians.
+ *
+ * @private
+ * @param {Feature<Point>} candidateMedian current candidate median
```

Comment:
I changed these params to `Feature<Point>` however, one param you are giving a `Feature<Point>` and the other (previousCandidate) you are giving an `Array<number>`...